streams/src/main/java/org/apache/kafka/streams/state/internals/KeyValueRawStoreWrapper.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.errors.InvalidStateStoreException;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.StateStoreContext;
+import org.apache.kafka.streams.processor.api.ProcessorContext;
+import org.apache.kafka.streams.query.Position;
+import org.apache.kafka.streams.query.PositionBound;
+import org.apache.kafka.streams.query.Query;
+import org.apache.kafka.streams.query.QueryConfig;
+import org.apache.kafka.streams.query.QueryResult;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.TimestampedKeyValueStore;
+import org.apache.kafka.streams.state.ValueAndTimestamp;
+import org.apache.kafka.streams.state.VersionedKeyValueStore;
+import org.apache.kafka.streams.state.VersionedRecord;
+
+/**
+ * A wrapper class for non-windowed key-value stores used within the DSL. All such stores are
+ * instances of either {@link TimestampedKeyValueStore} or {@link VersionedKeyValueStore}.
+ *
+ * @param <K> The key type
+ * @param <V> The value type
+ */
+public class KeyValueRawStoreWrapper<K, V> implements StateStore {
+
+    public static final long PUT_RETURN_CODE_IS_LATEST
+        = VersionedKeyValueStore.PUT_RETURN_CODE_VALID_TO_UNDEFINED;
+
+    private KeyValueStore<Bytes, byte[]> timestampedStore = null;
+    private VersionedKeyValueStore<Bytes, byte[]> versionedStore = null;
+
+    // same as either timestampedStore or versionedStore above. kept merely as a convenience
+    // to simplify implementation for methods which do not depend on store type.
+    private StateStore store = null;
+
+    @SuppressWarnings("unchecked")
+    public KeyValueRawStoreWrapper(final ProcessorContext<?, ?> context, final String storeName) {
+        try {
+            // first try timestamped store
+            timestampedStore = ((WrappedStateStore<KeyValueStore<Bytes, byte[]>, K, V>) ((WrappedStateStore<KeyValueStore<Bytes, byte[]>, K, V>) context.getStateStore(storeName)).wrapped()).wrapped();
```

Comment:
This is pretty much just hacked together right now... (Same in L67 below.)

Reply:
Just cycling back to this -- not sure what the difference between `RawXxxStore` and `WrappedKeyValueState` would be? (Hope you can still remember... 😁)